### PR TITLE
BMI: adjust git url to new repo location

### PR DESCRIPTION
**Pull request by carns**
Moving repo to new permanent location on github. Also renaming default branch.

### CODE REVIEW

File: var/spack/repos/builtin/packages/bmi/package.py [hunk 1]
```diff
@@ -9,12 +9,13 @@
...
 
-    homepage = 'https://xgitlab.cels.anl.gov/sds/bmi'
-    git = 'https://xgitlab.cels.anl.gov/sds/bmi.git'
+    homepage = 'https://github.com/radix-io/bmi/'
+    git = 'https://github.com/radix-io/bmi.git'
 
     maintainers = ['carns']
 
-    version('develop', branch='master')
+    version('develop', branch='main')
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/bmi/package.py]: Why are you specifying a `develop` version pointing to `main`?  (I realize you currently only have a `main` branch`.)

**Review by tldahlgren — COMMENTED** [review]
Confirmed the software is available at the new URL *and* noted that the old URL is archived/read-only.

File: var/spack/repos/builtin/packages/bmi/package.py (hunk 1 above)
Comment (carns) [reply var/spack/repos/builtin/packages/bmi/package.py]: I just pushed an update to this PR that removes that.  I had left it their for compatibility reasons, but I don't think it's really needed.

**Review by tldahlgren — APPROVED** [review]
LGTM.  Approval contingent on all required checks passing.

**Review by alalazo — APPROVED** [review]

**Pull request merged**